server/src/main/java/org/apache/uniffle/server/ShuffleServerGrpcService.java

```diff
@@ -528,6 +529,15 @@ public void getShuffleResult(
     String appId = request.getAppId();
     int shuffleId = request.getShuffleId();
     int partitionId = request.getPartitionId();
+    BlockIdLayout blockIdLayout = BlockIdLayout.DEFAULT;
+    // legacy clients might send request without block id layout, we fall back to DEFAULT then
```

Comment:
If using the latest client connecting to the older shuffle server, the incorrect blocks list will be returned. 

Maybe we should upgrade the default SHUFFLE_SERVER_VERSION to avoid this(fail fast is another option if we can), this looks a uncompatible change.

Reply:
Can the client somehow check if it is connected to an older shuffle server? It could then fail (only if the block id layout is different to DEFAULT).

Reply:
We don't need to  guarantee the new client connect the old shuffle server correctly. We don't consider the compatible issue in the other features.

Reply:
> We don't need to guarantee the new client connect the old shuffle server correctly. We don't consider the compatible issue in the other features.

But this will effect data correctness if someone missed this when upgrading

Reply:
> Can the client somehow check if it is connected to an older shuffle server?

Only the SHUFFLE_SERVER_VERSION will ensure the newer client won't be allocated to the older server.